### DISCUSSION TITLE
Channel url

### DISCUSSION THREAD

**Discussion by feadrick** [Resgrid/Core#42] [Q&A] [answered]
hi im writing this discussion ,so that i can have a clear picture on how to connect the bigboard with the resgrid core project.
there is a config that i'm not sure what value should i put .
i dont know which url i point for ChannelUrl config. i already setup the web and the api.
![config](https://user-images.githubusercontent.com/26769295/133961066-af7f3010-d4b2-4ede-a392-8d607ad00fb2.PNG)

**Comment by ucswift** [comment, marked as answer]
Hey @feadrick, the ChannelUrl is used for SignalR communications. This was the api url, but it's now being split into it's own project (https://github.com/Resgrid/Core/tree/master/Web/Resgrid.Web.Eventing) to allow for better control of multi server router, seeming it uses web sockets it needs have sticky routing whereas the API and Web projects should be load balanced differently.

At the moment that is not in the built release package, if you build the latest source you can deploy it as ASP.Net Core web app to Linux or Windows. I'm working publishing a docker image for it as well shortly. When you deploy that website it's the location you'll use for the ChannelUrl.

**Reply by feadrick** [reply]
so which means i need to deploy 3 site right?
https://github.com/Resgrid/Core/tree/master/Web/Resgrid.Web.Eventing= webevent(channelurl)
https://github.com/Resgrid/Core/tree/master/Web/Resgrid.WebCore = Web
https://github.com/Resgrid/Core/tree/master/Web/Resgrid.Web.ServicesCore = Api
is this correct?

**Reply by ucswift** [reply]
Yes there will be 3 sites, now in total that need to be deployed, you've got the correct ones.

For the API and Web if your using load balancing, you can use round robin, least connections or another alog. For Eventing (webevent) you'll need to use some form of sticky session/ip hash to ensure websocket connections go to the same server.
